### PR TITLE
Set default RxFactory in SyncConfiguration when RxJava is available.

### PR DESCRIPTION
**Pull request by zaki50**
@realm/java

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 2.1.1
+
+### bug fixes
```
Comment (cmelchior) [inline CHANGELOG.md]: "bug fixes" -> "Bug Fixes"

Comment (cmelchior) [inline CHANGELOG.md]: Actually this should probably be under the `Object Server API Changes` header just like for `2.1.0`

Comment (zaki50) [inline CHANGELOG.md]: done

**Comment by zaki50** [comment]
`Could not update commit status. Message: GitHub API rate limit exceeded`

I'll merge this.

**Pull request merged**